db/compaction/compaction.cc

```diff
@@ -373,6 +376,10 @@ Compaction::Compaction(
     }
   }
 
+  if (earliest_snapshot_.has_value()) {
+    FilterInputsForCompactionIterator();
```

Comment:
Do you mean to merge the logic of filling `input_levels_` into `FilterInputsForCompactionIterator` for this case? It will still do the same `DoGenerateLevelFilesBrief` call though. It think functional wise, it's not very different.

Reply:
I meant right now we would need to `DoGenerateLevelFilesBrief()` twice. Since `input_leves_` in only used for creating input iterator, we don't need a separate `filtered_non_start_input_level_` and we can just store filtered result directly in `input_levels_`.

Reply:
This is updated now.